lte/gateway/c/session_manager/SessionEvents.cpp

```diff
@@ -197,5 +222,25 @@ std::string EventsReporterImpl::get_mac_addr(const SessionConfig& config) {
   return mac_addr;
 }
 
+std::string EventsReporterImpl::get_imei(const SessionConfig& config) {
+  // IMEI is only relevant for LTE
+  const auto& rat_specific    = config.rat_specific_context;
+  std::string imei = "";
+  if (rat_specific.has_lte_context()) {
+    imei = rat_specific.lte_context().imei();
+  }
+  return imei;
+}
+
+std::string EventsReporterImpl::get_spgw_ipv4(const SessionConfig& config) {
```

Comment:
wouldn't it be easier to read this way?
`rat_specific` can be understood directly from `config.rat_specific_context`

```
std::string EventsReporterImpl::get_spgw_ipv4(const SessionConfig& config) {
  // IMEI is only relevant for LTE
  if (config.rat_specific_context.has_lte_context()) {
    return rat_specific.lte_context().spgw_ipv4();
  }
  return "";
}

```

or if you want it reverse maybe is cleaner?
```
std::string EventsReporterImpl::get_spgw_ipv4(const SessionConfig& config) {
  // IMEI is only relevant for LTE
  if (!config.rat_specific_context.has_lte_context()) {
    return "";
  }
  return rat_specific.lte_context().spgw_ipv4();
}

```